dev-docs/howto/vpn/templates/service.yaml

```diff
@@ -0,0 +1,25 @@
+apiVersion: v1
+kind: Service
+metadata:
+  name: {{ include "..fullname" . }}-lb
+  labels:
+    {{- include "..labels" . | nindent 4 }}
+spec:
+  type: LoadBalancer
+  selector:
+    {{- include "..selectorLabels" . | nindent 4 }}
+    component: frontend
+  ports:
+    {{- if .Values.ipsec.enabled }}
+    - name: isakmp
+      protocol: UDP
+      port: 500
+    - name: ipsec-nat-t
+      protocol: UDP
+      port: 4500
```

Comment:
Interestingly, this creates a LB with ports _from_ 500 _to_ 4500. We should either try to fix this with annotations (if possible) or just deploy 2 different LBs (this is more agnostic than cloud-specific annotations).

Reply:
This seems to be a bug in the load balancer implementation, doesn't it? https://kubernetes.io/docs/concepts/services-networking/service/#multi-port-services are a supported concept, and I only saw the note about needing to use the same protocol for all ports.

The correct workaround may be to just not expose 500, and configure on-prem to start with encap right away.

Reply:
I just double-checked, and while the LB frontend claims to be forwarding the 500-4500 range, only  500 and 4500 are observed on the node.

Reply:
https://github.com/kubernetes/cloud-provider-gcp/blob/master/providers/gce/gce_loadbalancer_external.go#L838
Oh well, this is not how I would have implemented that. 
And they even have a different logic for the firewall rules:
https://github.com/kubernetes/cloud-provider-gcp/blob/master/providers/gce/gce_loadbalancer_internal.go#L868C10-L868C10
This is truly their default way of doing things.

Reply:
I'm considering this resolved, then.